include/dfu_client.h

```diff
@@ -0,0 +1,191 @@
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+/**@file dfu_client.h
+ *
+ * @defgroup dfu_client DFU client to download and apply new firmware
+ * (or patches) to target. Currently, only HTTP protocol is supported for
+ * download.
+ * @{
+ * @brief DFU Client interface to connect to server, download and apply new
+ * firmware to a selected target.
+ */
+
+#ifndef DFU_CLIENT_H__
+#define DFU_CLIENT_H__
+
+#include <zephyr/types.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+struct dfu_client_object;
+
+/**@defgroup dfu_client_status DFU download status.
+ *
+ * @brief Indicates the status of download.
+ * @{
+ */
+enum dfu_client_status {
+	DFU_CLIENT_STATUS_IDLE      = 0x00,				/**< Indicates that the client was either
+													 * never connected to the server or is disconnected.
+													 */
+	DFU_CLIENT_STATUS_CONNECTED = 0x01,				/**< Indicates that the client is connected to the server,
+													 * and there is no on going download.
+													 */
+	DFU_CLIENT_STATUS_DOWNLOAD_INPROGRESS = 0x02,	/**< Indicates that the client is connected to the server,
+													 * and download is in progress.
+													 */
+	DFU_CLIENT_STATUS_DOWNLOAD_COMPLETE = 0x03,		/**< Indicates that the client is connected to the server,
+													 * and download is in complete.
+													 */
+	DFU_CLIENT_STATUS_HALTED = 0x04					/**< Indicates that the firmware download is halted by the application.
+													 * This status indicates that the application indicated a failure when
+													 * handling the @ref DFU_CLIENT_EVT_DOWNLOAD_FRAG event.
+													 */
+};
+/* @} */
+
+/**@defgroup dfu_client_evt DFU events.
+ * @brief The DFU Events.
+ * @{
+ */
+enum dfu_client_evt {
+	DFU_CLIENT_EVT_ERROR = 0x00,			/**< Indicates error during download.
+											 * The application should disconnect and retry on reception of this event and reconnect.
+											 */
+	DFU_CLIENT_EVT_DOWNLOAD_FRAG = 0x01,	/**< Indicates reception of a fragment of firmware during download.
+											 * The fragment field of the dfu object @ref dfu_client_object points to the firmware fragment
+											 * and the fragment size indicates the size of the fragment.
+											 */
+	DFU_CLIENT_EVT_DOWNLOAD_DONE = 0x02,	/**< Indicates that the download is complete. */
+};
+/* @} */
+
+/**@brief DFU client event handler.
+ *
+ * @details The application is notified of the status of firmware download using this event handler.
+ *          The application can use the return value to indicate if it handled the event successfully or not.
+ *          This feedback is useful when, for example, a fault fragment was received or the application
+ *          was unable to process a fimrware fragment.
+ *
+ * @param[in] dfu 		The DFU instance.
+ * @param[in] event     The event.
+ * @param[in] status    Event status, can be 0 or an errno value.
+ *
+ * @retval 0 indicates the event was handled successfully.
```

Comment:
You can use ```@return``` here as ```@retval``` is used to create a list of return values.

```
@retval 0 indicates the event was handled successfully
@retval !=0 indicates the application failed to handle the event
```

Reply:
No - @retval introduces a specific value (like 0). It doesn't matter if it's a list of values or only one.